components/entitlement/org.wso2.carbon.identity.entitlement.endpoint/src/main/java/org/wso2/carbon/identity/entitlement/endpoint/util/JSONResponseWriter.java

```diff
@@ -115,13 +134,230 @@ private static JsonObject abstractResultToJSONObject(AbstractResult result) thro
             jsonResult.add(EntitlementEndpointConstants.ASSOCIATED_ADVICE, advices);
         }
 
-        /**
-         * Todo: Category, PolicyIdentifierList
-         */
+        // If includeInResponse=true, other attributes will be populated from here with the decision.
+        if (((Result) result).getAttributes() != null && !((Result) result).getAttributes().isEmpty()) {
+            Set<Attributes> attributes = ((Result) result).getAttributes();
+
+            for (Attributes attribute : attributes) {
+
+                switch (attribute.getCategory().toString()) {
+                    case EntitlementEndpointConstants.CATEGORY_ACTION_URI:
+
```

Comment:
It'd be better if we can remove extra lines. Follow the formatting you've used in [JSONRequestParser.java](https://github.com/wso2/carbon-identity-framework/pull/1492/files#diff-d79495a4b40d533991fce95c2d3ad587)